debezium-reactive/src/test/java/io/debezium/reactive/ReactiveEngineTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.reactive;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.debezium.reactive.converters.AsSourceRecord;
+import io.debezium.util.Testing;
+import io.reactivex.Flowable;
+
+public class ReactiveEngineTest extends AbstractReactiveEngineTest {
```

Comment:
When running this test, it passes, but I'm seeing this in the log:

```
java.lang.IllegalStateException: An event was not committed after it was processed by pipeline. The pipeline processing has failed or method commit() was not called.
	at io.debezium.reactive.internal.ReactiveEngineImpl.lambda$2(ReactiveEngineImpl.java:126)
	at io.reactivex.internal.operators.flowable.FlowableInternalHelper$SimpleBiGenerator.apply(FlowableInternalHelper.java:62)
	at io.reactivex.internal.operators.flowable.FlowableInternalHelper$SimpleBiGenerator.apply(FlowableInternalHelper.java:53)
	at io.reactivex.internal.operators.flowable.FlowableGenerate$GeneratorSubscription.request(FlowableGenerate.java:109)
	at io.reactivex.internal.subscribers.LambdaSubscriber.request(LambdaSubscriber.java:114)
	at io.reactivex.internal.operators.flowable.FlowableInternalHelper$RequestMax.accept(FlowableInternalHelper.java:220)
	at io.reactivex.internal.operators.flowable.FlowableInternalHelper$RequestMax.accept(FlowableInternalHelper.java:216)
	at io.reactivex.internal.subscribers.LambdaSubscriber.onSubscribe(LambdaSubscriber.java:52)
	at io.reactivex.internal.operators.flowable.FlowableGenerate.subscribeActual(FlowableGenerate.java:52)
	at io.reactivex.Flowable.subscribe(Flowable.java:13234)
	at io.reactivex.Flowable.subscribe(Flowable.java:13170)
	at io.reactivex.Flowable.subscribe(Flowable.java:13126)
	at io.debezium.reactive.ReactiveEngineTest.errorInConsumer(ReactiveEngineTest.java:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
@gunnarmorling It is a result of one of negative tests